preview/MsixCore/msixmgr/UnpackProvider.cpp

```diff
@@ -0,0 +1,184 @@
+#include "MSIXWindows.hpp"
+#include "UnpackProvider.hpp"
+#include "ApplyACLsProvider.hpp"
+
+#include <TraceLoggingProvider.h>
+#include "msixmgrLogger.hpp"
+#include "..\msixmgrLib\GeneralUtil.hpp"
+#include "MSIXWindows.hpp"
+#include <shlobj_core.h>
+#include <CommCtrl.h>
+#include <map>
+#include <iostream>
+
+using namespace MsixCoreLib;
+using namespace std;
+
+namespace MsixCoreLib
+{
+
+    HRESULT UnpackPackage(
+        _In_ std::wstring packageFilePath,
+        _In_ std::wstring destination,
+        _In_ bool isApplyACLs)
+    {
+        MSIX_PACKUNPACK_OPTION unpackOption = MSIX_PACKUNPACK_OPTION_UNPACKWITHFLATSTRUCTURE;
+        MSIX_VALIDATION_OPTION validationOption = MSIX_VALIDATION_OPTION_FULL;
+
+        auto unpackDestinationUTF8 = utf16_to_utf8(destination);
+        std::vector<char> unpackDestination(unpackDestinationUTF8.c_str(), unpackDestinationUTF8.c_str() + unpackDestinationUTF8.size() + 1);
+
+        ComPtr<IAppxFactory> factory;
+        RETURN_IF_FAILED(CoCreateAppxFactoryWithHeap(MyAllocate, MyFree, validationOption, &factory));
+
+        ComPtr<IStream> stream;
+        RETURN_IF_FAILED(CreateStreamOnFileUTF16(packageFilePath.c_str(), true, &stream));
+
+        ComPtr<IAppxPackageReader> reader;
+        HRESULT hrCreatePackageReader = factory->CreatePackageReader(stream.Get(), &reader);
+
+        if (FAILED(hrCreatePackageReader))
+        {
+            std::wcout << std::endl;
+            std::wcout << L"Failed with HRESULT " << hrCreatePackageReader << L" when trying to create package reader for " << packageFilePath << L". Please confirm that the certificate for the package has been installed." << std::endl;
+            return hrCreatePackageReader;
+        }

```

Comment:
this should be a specific error code. (or at least your warning about the certificate not being installed should only be displayed when that error code is encountered). I would prefer to do the wcout in msixmgr.cpp  for the specific error code encountered so you don't see multiple error messages when it fails here.